flink-python/pyflink/table/expressions.py

```diff
@@ -544,6 +544,25 @@ def if_then_else(condition: Union[bool, Expression[bool]], if_true, if_false) ->
     return _ternary_op("ifThenElse", condition, if_true, if_false)
 
 
+def coalesce(*args) -> Expression:
+    """
+    Coalesce specifies a series of expressions, and returns the first expression whose value is
+    not null. If all the expressions evaluate as null, coalesce returns a null value. The return
+    type of the coalesce function is the least common type among its arguments, and it's nullable
+    if all its arguments are nullable.
+
+    e.g. coalesce(null, "-") returns "-".
+
+    e.g. coalesce(col("f0"), col("f1"), "-") returns the first non-null value among f0 and f1, if both
+    are null returns "-".
+
+    :param args: the input expressions.
+    """
+    gateway = get_gateway()
+    args = to_jarray(gateway.jvm.Object, [_get_java_expression(arg) for arg in args])
```

Comment:
This is fine, I guess, but after we merge https://github.com/apache/flink/pull/17186/files#diff-5930126827bbe9069974e6094238fca02034cab5435aafd5a6fc012b34f9df83R70 we'd also be able to just call `_varargs_op("coalesce", *args)`, I think.